src/main/drivers/serial_tcp.c

```diff
@@ -122,13 +128,13 @@ serialPort_t *serTcpOpen(int id, serialReceiveCallbackPtr rxCallback, void *rxCa
 {
     tcpPort_t *s = NULL;
 
-#if defined(USE_UART1) || defined(USE_UART2) || defined(USE_UART3) || defined(USE_UART4) || defined(USE_UART5) || defined(USE_UART6) || defined(USE_UART7) || defined(USE_UART8)
-    if (id >= 0 && id < SERIAL_PORT_COUNT) {
-    s = tcpReconfigure(&tcpSerialPorts[id], id);
+    if (id > SERIAL_PORT_NONE && id < SERIAL_PORT_COUNT) {
```

Comment:
(don't assume that SERIAL_PORT_NONE == -1; this code is wrong, it will refuse LPUART, but allocated space for all ports, including VCP and SOFTSERIAL
```suggestion
    if (id >= 0 && id < SERIAL_PORT_COUNT) {
```

Reply:
The same argument can be made for not assuming the index it starts with is 0. I have added a self explaining name for this value (instead of 0 it is SERIAL_PORT_START_INDEX).

Reply:
I am not sure it will refuse LPUART? serial port count should be last index + 1. Agree it is superfluous to add in support for VCP and softserial, however given it is only used by the SITL at the moment there is little lost.